Description: Disable jemalloc tests and on s390x
Author: Ximin Luo <infinity0@debian.org>
Applied-Upstream: commit:1a2ed98d344b6cbddc57db8841b42f935877e08d
---
This patch header follows DEP-3: http://dep.debian.net/deps/dep3/
Index: rust/src/librustc_back/target/s390x_unknown_linux_gnu.rs
===================================================================
--- rust.orig/src/librustc_back/target/s390x_unknown_linux_gnu.rs	2017-02-26 18:08:22.891629489 +0100
+++ rust/src/librustc_back/target/s390x_unknown_linux_gnu.rs	2017-02-26 18:08:22.887629473 +0100
@@ -19,6 +19,8 @@
     // Pass the -vector feature string to LLVM to respect this assumption.
     base.features = "-vector".to_string();
     base.max_atomic_width = Some(64);
+    // see #36994
+    base.exe_allocation_crate = "alloc_system".to_string();
 
     Ok(Target {
         llvm_target: "s390x-unknown-linux-gnu".to_string(),
Index: rust/src/test/compile-fail/allocator-rust-dylib-is-jemalloc.rs
===================================================================
--- rust.orig/src/test/compile-fail/allocator-rust-dylib-is-jemalloc.rs	2017-02-26 18:08:22.891629489 +0100
+++ rust/src/test/compile-fail/allocator-rust-dylib-is-jemalloc.rs	2017-02-26 18:08:22.887629473 +0100
@@ -29,9 +29,11 @@
 // ensure we get the same error.
 //
 // So long as we CI linux/OSX we should be good.
-#[cfg(any(target_os = "linux", target_os = "macos"))]
+#[cfg(any(all(target_os = "linux", any(target_arch = "x86", target_arch = "x86_64")),
+          target_os = "macos"))]
 extern crate alloc_system;
-#[cfg(not(any(target_os = "linux", target_os = "macos")))]
+#[cfg(not(any(all(target_os = "linux", any(target_arch = "x86", target_arch = "x86_64")),
+              target_os = "macos")))]
 extern crate allocator1;
 
 fn main() {
Index: rust/src/test/run-pass/allocator-default.rs
===================================================================
--- rust.orig/src/test/run-pass/allocator-default.rs	2017-02-26 18:08:22.891629489 +0100
+++ rust/src/test/run-pass/allocator-default.rs	2017-02-26 18:08:22.887629473 +0100
@@ -10,7 +10,8 @@
 
 #![feature(alloc_jemalloc)]
 
-#[cfg(any(target_os = "linux", target_os = "macos"))]
+#[cfg(any(all(target_os = "linux", any(target_arch = "x86", target_arch = "x86_64")),
+          target_os = "macos"))]
 extern crate alloc_jemalloc;
 
 fn main() {
